From cc06e4d3f704a472e898cf179ebcc269463416db Mon Sep 17 00:00:00 2001
From: Igor Gnatenko <i.gnatenko.brain@gmail.com>
Date: Fri, 11 Oct 2013 03:09:27 +0400
Subject: [PATCH 1/2] build jthread static

Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>
---
 src/jthread/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/jthread/CMakeLists.txt b/src/jthread/CMakeLists.txt
index be38f87..91cc5e5 100644
--- a/src/jthread/CMakeLists.txt
+++ b/src/jthread/CMakeLists.txt
@@ -6,7 +6,7 @@ else( UNIX )
 	set(jthread_platform_LIBS "")
 endif( UNIX )
 
-add_library(jthread ${jthread_SRCS})
+add_library(jthread STATIC ${jthread_SRCS})
 
 target_link_libraries(
 	jthread
-- 
1.8.3.1

